code/modules/client/preferences.dm

```diff
@@ -1318,6 +1319,9 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 			if(initial(T.mood_quirk) && CONFIG_GET(flag/disable_human_mood))
 				lock_reason = "Mood is disabled."
 				quirk_conflict = TRUE
+			if((Test.allowed_species && !(pref_species.id in Test.allowed_species)) || (Test.blocked_species && (pref_species.id in Test.blocked_species)))
```

Comment:
There has to be a check for if there *is* anything in allowed_species because if it has nothing it simply blocks every species, and having a list of every roundstart playable species id in the list define alongside forcing people to add the species id to a list in _quirks whenever they add a new one is a bit silly
The check for if there is anything in blocked_species can be removed though that is fair it doesn't actually achieve anything and is kinda just a waste

Reply:
`(Test.allowed_species && !(pref_species.id in Test.allowed_species)`
and 
`!(pref_species.id in Test.allowed_species)`

will always return the same thing.

Reply:
It doesn't, first check checks if the list isn't empty and the second checks if it doesn't have the species id in it
I've literally already tested it before, there's a reason I coded it like this
If the list is empty the list will never have the species id and so will always return TRUE to the second operation
But if the first one returns FALSE it ignores that because if there isn't anything in the list that means the list wasn't overriden by the quirk and is allowed to everyone
You can test it if you want.

Reply:
Actually yeah im wrong on that one. Whoopsie.

That does beg the question, why are you adding both a whitelist and a blacklist instead of just opting for a blacklist?

Reply:
Mostly convenience, there's a lot of species ids you have to add if you want to have a quirk blocked only for one or two species like IPCs being unable to have blood deficiency as it has no effect on them
Should this be handled elsewhere? Absolutely
Do I have the time to make a separate system for that or even just attach it to a new prefs menu area? Sadly no, so just reusing the quirk framework for this is easier, you only need to redefine the one to which you need to add the least species ids
This is here to allow for things like the vampirism quirk (the reason I even did it in the first place, it's gor a very weird position in balance but is currently neutral so people can try it along some other things) and even more species-specific(as in present in less species) quirks
Later I should probably make it hide a few unavailable quirks if the quirk window at some point gets far too bloated before we get a customization system proper (something that shouldn't be happening but could)

Reply:
make a macro for ALL_SPECIES and then just do

blacklist = ALL SPECIES - desired_species

Reply:
Well, as I've said a few times I'm not a position for much work I can only do things directly on the github website/app and I won't be able to do stuff properly until at most the latter half of January, I'm just maintaining this one PR to keep it functional and not conflciting with other important stuff
That was planned but with my workplace non-functional I can't do as much and what is right now works